src/fileloader.h

```diff
@@ -105,6 +108,8 @@ class FileLoader : public QObject
                     this, SIGNAL(loadProgress(int)));
             connect(geom.get(), SIGNAL(loadStepStarted(QString)),
                     this, SIGNAL(loadStepStarted(QString)));
+            connect(geom.get(), SIGNAL(resetProgress()),
+                    this, SIGNAL(resetProgress()));
```

Comment:
Thanks for having a look at this.  I left it in a bit of a mess after some refactoring a while back which really improved some other things but messed up the loading bar.

It would be simpler just to emit `resetProgress` directly from `FileLoader` after invoking `Geometry::loadFile`, rather than involving the Geometry class at all.  If you think it makes a lot more sense to have the event emitted by `Geometry`, I'd rather call it `Geometry::loadFinished` or something.

In the process of looking at this I've noticed I never disconnected the signals from `geom` after the load.  This is a hangover from the way it used to work where the FileLoader was deleted after loading a group of files.  The `FileLoader` is now persistent so we should probably disconnect the signals explicitly to avoid the number of connections growing without bound as files are loaded and unloaded.